scripts/pythonscripts/sync_postgres_to_mssql.py

```diff
@@ -0,0 +1,228 @@
+            chunk = chunk.where(pd.notnull(chunk), None)
+            
+            # Fast batch insert into temp table
+            sql_cursor.fast_executemany = True
+            sql_cursor.executemany(insert_sql.replace(f"kawasaki.{table_name}", temp_table_name), 
+                                 chunk.values.tolist())
+            sql_conn.commit()
+            
+            # Move data from temp to final table in smaller batches
+            sql_cursor.execute(f"""
+                INSERT INTO kawasaki.{table_name}
+                SELECT TOP ({len(chunk)}) * FROM {temp_table_name};
+                DELETE TOP ({len(chunk)}) FROM {temp_table_name};
+            """)
+            sql_conn.commit()
+            
+            rows_copied += len(chunk)
+            chunk_time = time.time() - chunk_start
+            rows_per_second = len(chunk) / chunk_time if chunk_time > 0 else 0
+            
+            print(f"Progress for {table_name}: {rows_copied:,}/{total_rows:,} rows "
+                  f"({(rows_copied/total_rows*100):.1f}%) - "
+                  f"Speed: {rows_per_second:.0f} rows/sec")
+
+        # Drop temp table
+        sql_cursor.execute(f"DROP TABLE {temp_table_name}")
+        sql_conn.commit()
+
+        end_time = time.time()
+        total_time = end_time - start_time
+        avg_speed = rows_copied / total_time if total_time > 0 else 0
+        print(f"Completed copying {table_name}: {rows_copied:,} rows in {total_time:.2f} seconds "
+              f"(avg {avg_speed:.0f} rows/sec)")
+        
+        # Verify row count
+        sql_cursor.execute(f"SELECT COUNT(*) FROM kawasaki.{table_name}")
+        final_count = sql_cursor.fetchone()[0]
+        print(f"Final row count in destination table {table_name}: {final_count:,}")
+        if final_count != total_rows:
+            print(f"WARNING: Row count mismatch for {table_name}! Source: {total_rows:,}, Destination: {final_count:,}")
+
+    except Exception as e:
+        print(f"Error copying table {table_name}: {str(e)}")
+        import traceback
+        print(traceback.format_exc())
+    finally:
+        # Cleanup temp table if it exists
+        try:
+            sql_cursor.execute(f"IF OBJECT_ID('tempdb..{temp_table_name}') IS NOT NULL DROP TABLE {temp_table_name}")
+            sql_conn.commit()
+        except:
+            pass
+            
+        if 'cursor' in locals():
+            cursor.close()
+        if 'sql_cursor' in locals():
+            sql_cursor.close()
+        if 'pg_conn' in locals():
+            pg_conn.close()
+        if 'sql_conn' in locals():
+            sql_conn.close()
+
+def main():
+    tables = get_schema_tables()
+    print(f"Found {len(tables)} tables to copy")
+    
+    # ThreadPoolExecutor to copy tables in parallel
+    with ThreadPoolExecutor(max_workers=40) as executor:
+        executor.map(copy_table, tables)
+    
+    print("Schema copy completed")
+
+if __name__ == "__main__":
+    main()
```

Comment:
Do we need a `main()` fuction? Could you just copy all code under main() under line 228?